src/time.h

```diff
@@ -48,6 +50,97 @@ std::tm safe_gmtime(const std::time_t* t);
 
 }
 
+// Abstract class for an expiration.
+class Expiration {
+ public:
+  virtual ~Expiration() = default;
+  virtual bool IsExpired() = 0;
+  virtual void Reset(std::chrono::seconds duration) = 0;
+};
+
+// Implementation of an expiration paramterized over a clock type.
+template<typename Clock>
+class ExpirationImpl : public Expiration {
+ public:
+  static std::unique_ptr<Expiration> New(std::chrono::seconds slack) {
```

Comment:
Also, you need to `#include <memory>` to get `unique_ptr`.

Reply:
Done.